drivers/serial/pty.c

```diff
@@ -452,17 +452,26 @@ static ssize_t pty_read(FAR struct file *filep, FAR char *buffer, size_t len)
   FAR struct pty_dev_s *dev;
   ssize_t ntotal;
 #ifdef CONFIG_SERIAL_TERMIOS
-  ssize_t nread;
-  size_t i;
-  char ch;
-  int ret;
+  ssize_t i = 0;
```

Comment:
```suggestion
  size_t i = 0;
```
since this is a positive counter

Reply:
But ntotal is ssize_t, and we do i < ntotal in for loop.

Reply:
ok. So this is just to avoid bug with signed t unsigned compare.